tgui/packages/tgui-say/handlers/incrementChannel.tsx

```diff
@@ -24,9 +36,19 @@ export const handleIncrementChannel = function (this: Modal) {
       // Disables thinking indicator for OOC channel
       this.timers.channelDebounce({ mode: false });
     }
-    this.setState({
-      buttonContent: CHANNELS[channel + 1],
-      channel: channel + 1,
-    });
+    if (
+      channel === ADMIN_CHANNEL - 1 &&
+      CHANNELS.length - 1 !== ADMIN_CHANNEL
+    ) {
+      this.setState({
+        buttonContent: CHANNELS[channel + 2],
+        channel: channel + 2,
+      });
```

Comment:
Instead of this, you should have a set of names to blacklist. Then, add 1 to channel until you get something not on the blacklist, or wrap around to the beginning (where you will continue to check until you get a valid result)

Reply:
Can I add the same behavior for OOC then? so it can only cycle within the IC buttons? If I am going to rework it into a blacklist anyway.